partner_location_auto_create/models/res_partner.py

```diff
@@ -0,0 +1,137 @@
+# -*- encoding: utf-8 -*-
+##############################################################################
+#
+#    OpenERP, Open Source Management Solution
+#    This module copyright (C) 2015 Savoir-faire Linux
+#    (<http://www.savoirfairelinux.com>).
+#
+#    This program is free software: you can redistribute it and/or modify
+#    it under the terms of the GNU Affero General Public License as
+#    published by the Free Software Foundation, either version 3 of the
+#    License, or (at your option) any later version.
+#
+#    This program is distributed in the hope that it will be useful,
+#    but WITHOUT ANY WARRANTY; without even the implied warranty of
+#    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#    GNU Affero General Public License for more details.
+#
+#    You should have received a copy of the GNU Affero General Public License
+#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+##############################################################################
+
+from openerp import models, fields, api, _
+
+
+class ResPartner(models.Model):
+    _inherit = 'res.partner'
+
+    @api.one
+    @api.depends('location_ids')
+    def count_locations(self):
+        self.locations_count = len(self.location_ids)
+
+    locations_count = fields.Integer(compute='count_locations', store=False)
+
+    location_ids = fields.One2many(
+        'stock.location', 'partner_id', string='Locations')
+
+    @api.multi
+    def button_locations(self):
+        self.ensure_one()
+
+        res = {
+            'name': _('Locations'),
+            'type': 'ir.actions.act_window',
+            'res_model': 'stock.location',
+            'view_type': 'form',
+        }
+
+        if len(self.location_ids) == 1:
+            res['res_id'] = self.location_ids[0].id
+            res['view_mode'] = 'form'
+        else:
+            res['domain'] = [('partner_id', '=', self[0].id)]
```

Comment:
With `self.ensure_one()`, `self[0].id` becomes redundant and can be simplified to `self.id`